main.sh

```diff
@@ -22,8 +22,10 @@ detect_build_necessity() {
   IFS=' ' read -r -a keywords <<< "$keywords_str"
 
   case $commit_msg in
-    --|' ') commit_msg="$(git log --format=%B -n 1 HEAD)" ;;
```

Comment:
idea is that, it can be provided via input from github apis. like:

```
${{ github.event.workflow_run.head_commit.message }}
```

Reply:
https://docs.github.com/en/webhooks/webhook-events-and-payloads#push
pull_request does not have it. https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request